src/Hosting/Hosting/src/Internal/WebHost.cs

```diff
@@ -129,6 +129,10 @@ public void Initialize()
 
                 _applicationServicesException = ExceptionDispatchInfo.Capture(ex);
             }
+
+            // resolve configuration explicitly to mark it as resolved within the service
+            // provider, ensuring it will be properly disposed with the provider
+            _ = _applicationServices.GetService<IConfiguration>();
```

Comment:
This could still happen in WebHostBuilder just after calling Initialize.
https://github.com/aspnet/AspNetCore/blob/b8317c9f768460b509e95925a65d3ab78feba779/src/Hosting/Hosting/src/WebHostBuilder.cs#L178